### PR TITLE
copybook bugfixes

### PR DESCRIPTION
**Pull request by lxpdd**
Approached issues: #1516, #1533 and #DE533540

### CODE REVIEW

**Review by ishche — APPROVED** [review]

**Comment by lxpdd** [comment]
Manual tests failed. Work is still in progress

**Review by ap891843 — COMMENTED** [review]
GraalVM native build fails for this PR. It's still under investigation. Let's not merge it, until we fix the issue.

**Review by ap891843 — CHANGES_REQUESTED** [review]
Let's wait until we fix the native build. No changes are required from @lxpdd

**Review by ap891843 — APPROVED** [review]

**Comment by VitGottwald** [comment]
Does not work on internal test suite. Will send details in a private channel.

File: clients/cobol-lsp-vscode-extension/src/services/copybook/LocalCopybookResolver.ts [hunk 1]
```diff
@@ -32,10 +32,12 @@ export class LocalCopybookResolver implements CopybookResolver {
     private resolveURIList(list: string[]): string[] {
         const result: Set<string> = new Set<string>();
         list.filter(resource => resource !== "*").forEach(resource => {
-            const URI: URL = getURIFromResource(resource);
-            if (URI !== undefined) {
-                result.add(URI.href);
-            }
+            const uris: URL[] = getURIFromResource(resource);
+            uris.forEach(uri => {
+                if (uri != undefined) {
```
Comment (ishche) [inline clients/cobol-lsp-vscode-extension/src/services/copybook/LocalCopybookResolver.ts:37]: Can it actually be undefined?

File: clients/cobol-lsp-vscode-extension/src/services/util/FSUtils.ts [hunk 2]
```diff
@@ -45,20 +47,49 @@ export function getURIFrom(folder: string, entityName: string, extensions?: stri
     }
 }
 
+
+function posixPath(windowsPath) {
```
Comment (ishche) [inline clients/cobol-lsp-vscode-extension/src/services/util/FSUtils.ts:51]: I don't see where it is used. Is it?

File: clients/cobol-lsp-vscode-extension/src/services/util/FSUtils.ts [hunk 3]
```diff
@@ -45,20 +47,49 @@ export function getURIFrom(folder: string, entityName: string, extensions?: stri
...
  */
-export function getURIFromResource(resource: string): urlUtil.URL {
-    for (const workspaceFolder of SettingsUtils.getWorkspacesURI()) {
-        const uri: urlUtil.URL = (path.resolve(resource) === path.normalize(resource))
-            ? urlUtil.pathToFileURL(resource) :
-            new urlUtil.URL(path.normalize(path.join(workspaceFolder, resource)));
-        if (fs.existsSync(uri)) {
-            return uri;
+export function getURIFromResource(resource: string): urlUtil.URL[] {
+    let uris: urlUtil.URL[] = [];
```
Comment (ishche) [inline clients/cobol-lsp-vscode-extension/src/services/util/FSUtils.ts:64]: minor: should be const as on line 76

File: server/engine/src/main/java/org/eclipse/lsp/cobol/service/utils/WorkspaceFileService.java [hunk 4]
```diff
@@ -59,6 +65,14 @@ public String decodeURI(@NonNull String uri) {
     }
   }
 
+  @Override
+  public boolean isUriAbsolute(@NonNull final String uri) {
+    String cleanedUpUri = uri.replaceFirst("file:", "");
+    String[] schemeParseResult = cleanedUpUri.split(":");
```
Comment (ishche) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/service/utils/WorkspaceFileService.java:71]: Will it work for unix paths?

**Review by ishche — CHANGES_REQUESTED** [review]
Please, resolve conflicts with the development branch.

File: clients/cobol-lsp-vscode-extension/src/services/copybook/LocalCopybookResolver.ts (hunk 1 above)
Comment (lxpdd) [reply clients/cobol-lsp-vscode-extension/src/services/copybook/LocalCopybookResolver.ts:37]: have no idea, I'd rather not to change it in this PR

File: clients/cobol-lsp-vscode-extension/src/services/util/FSUtils.ts (hunk 3 above)
Comment (lxpdd) [reply clients/cobol-lsp-vscode-extension/src/services/util/FSUtils.ts:64]: fixed

File: clients/cobol-lsp-vscode-extension/src/services/util/FSUtils.ts (hunk 2 above)
Comment (lxpdd) [reply clients/cobol-lsp-vscode-extension/src/services/util/FSUtils.ts:51]: removed

**Review by zimlu02 — CHANGES_REQUESTED** [review]
All tests are OK!
The only detail to fix - please accept also '*' at the end of the path without ending '/'. Like this:
"slickoil/SLICK/SLICK/*"

**Review by ishche — APPROVED** [review]

**Pull request closed without merging**